### PR TITLE
Upgrades release smoke tests to use new CLI

### LINKED ISSUE #3277: Add new CLI versions of quick tests in CI tests
### Summary

The CI tests produced artifacts with some quick test using the legacy CLI -- add versions using the new CLI

### Background and Motivation

Making sure the CI tests stay current with the CLI changes

### Proposed Feature

The CI tests produced artifacts with some quick test using the legacy CLI -- add versions using the new CLI

### Alternatives

_No response_

### LINKED ISSUE #3679: Update quicktest.sh in workflows
### Summary

_No response_

### Background and Motivation

The workflows have some smoke tests that check whether the releases actually work on very simple tasks.
These are a bit obsolete and duplicative.

### Proposed Feature

Update quicktest.sh to use the new CLI and do all the smoke testing.
Remove the inline tests from the workflows.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #3277. Fixes #3679.

Changes the tests of release downloads to use the new CLI
Puts all such tests in quicktests.sh and not inline in the workflow

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Scripts/dafny [hunk 1]
```diff
@@ -8,7 +8,8 @@ while [ -h "$SOURCE" ]; do
     SOURCE="$(readlink "$SOURCE")"
     [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE"
 done
-DAFNY_ROOT="$( cd -P "$( dirname $( dirname "$SOURCE" ))" && pwd )"
+DIR=$(dirname "$SOURCE")
+DAFNY_ROOT=$( cd -P "$DIR" && pwd )/..
```
Comment (keyboardDrummer) [inline Scripts/dafny:12]: What has changed here?

Comment (davidcok) [reply Scripts/dafny:12]: Better handling of spaces in paths

File: .github/workflows/release-downloads-nuget.yml
Comment (atomb) [inline .github/workflows/release-downloads-nuget.yml:1]: This is a distinct improvement!

File: Scripts/quicktest.out [hunk 2]
```diff
@@ -9,19 +10,40 @@ Compiling on C#
 
 Dafny program verifier finished with 0 verified, 0 errors
 (42, 131)
-Compiling to Javascript
+Compiling on Javascript
```
Comment (atomb) [inline Scripts/quicktest.out:13]: This is a somewhat unfortunate change in wording, IMO.

**Review by atomb — DISMISSED** [review]

File: Scripts/quicktest.out (hunk 2 above)
Comment (davidcok) [reply Scripts/quicktest.out:13]: I'm not sure where my brain was -- I agree.

**Review by atomb — APPROVED** [review]
Yes, that's a nice improvement. :)

**Pull request merged**